apps/juxtaposition-ui/src/services/juxt-web/routes/console/feed.tsx

```diff
@@ -0,0 +1,167 @@
+import express from 'express';
+import { z } from 'zod';
+import { database } from '@/database';
+import { getCommunityHash } from '@/util';
+import { POST } from '@/models/post';
+import { config } from '@/config';
+import { parseReq } from '@/services/juxt-web/routes/routeUtils';
+import { WebGlobalFeedView, WebPersonalFeedView } from '@/services/juxt-web/views/web/feed';
+import { buildContext } from '@/services/juxt-web/views/context';
+import { WebPostListView } from '@/services/juxt-web/views/web/postList';
+import { CtrGlobalFeedView, CtrPersonalFeedView } from '@/services/juxt-web/views/ctr/feed';
+import { CtrPostListView } from '@/services/juxt-web/views/ctr/postList';
+import { PortalGlobalFeedView, PortalPersonalFeedView } from '@/services/juxt-web/views/portal/feed';
+import { PortalPostListView } from '@/services/juxt-web/views/portal/postList';
+
+export const feedRouter = express.Router();
+
+feedRouter.get('/', async function (req, res) {
+	const { auth, query } = parseReq(req, {
+		query: z.object({
+			pjax: z.string().optional()
+		})
+	});
+	const userContent = await database.getUserContent(auth().pid);
```

Comment:
I don't think I really like this pattern.. it would be very easy to accidentally not call `auth` when making a handler that happens to not reference userContent. Having it inside a parameter makes it look less important even though this is a load-bearing permissions check

Reply:
The authentication middleware is still applied, it won't get to the route without being authenticated. My intent was for `auth()` to be basically just a getter with an extra assertion so typescript is happy.

Reply:
Ok, fair enough